relay-cardinality/src/redis/cache.rs

```diff
@@ -129,9 +137,37 @@ impl<'a> CacheUpdate<'a> {
 }
 
 /// Critical section of the [`Cache`].
-#[derive(Debug, Default)]
+#[derive(Debug)]
 struct Inner {
     cache: hashbrown::HashMap<QuotaScoping, ScopedCache>,
+    last_vacuum: UnixTimestamp,
+}
+
+impl Inner {
+    fn vacuum(&mut self, ts: UnixTimestamp) {
+        // Debounce the vacuuming.
+        let secs_since_last_vacuum = ts.as_secs().saturating_sub(self.last_vacuum.as_secs());
```

Comment:
Why are we using a virtual "now" time here? Shouldn't the debounce depend on the actual current time?

nit: If `timestamp` is just a buffered now time, should we rename it to `now`, or `current_time`?

Reply:
Effectively it is always `now`, I was considering taking the `abs(ts - last_vacuum)` here, but since we only vacuum values which are in the past, based on the passed timestamp, I think it makes sense to only consider "the future".

Additionally using the passed in param also makes it a lot easier to test.